helpers/setup-spark-k8s.py

```diff
@@ -0,0 +1,126 @@
+
+def select_context_id(kube_cfg: Dict) -> int:
+    NO_CONTEXT = -1
+    SINGLE_CONTEXT = 0
+    context_names = [n['name'] for n in kube_cfg['contexts']]
+
+    selected_context_id = NO_CONTEXT
+    if len(context_names) == 1:
+        return SINGLE_CONTEXT
+
+    context_list_ux = '\n'.join(["{}. {}".format(context_names.index(a), a) for a in context_names])
+    while selected_context_id == NO_CONTEXT:
+        print (context_list_ux)
+        print('\nPlease select a kubernetes context by index:')
+        selected_context_id = input()
+
+        try:
+           int(selected_context_id)
+        except ValueError:
+            print('Invalid context index selection, please try again....')
+            selected_context_id = NO_CONTEXT
+            continue
+
+        if int(selected_context_id) not in range(len(context_names)):
+            print('Invalid context index selection, please try again....')
+            selected_context_id = NO_CONTEXT
+            continue
+
+    return int(selected_context_id)
+
+def get_defaults_from_kubeconfig(kubeconfig: str, context: str = None) -> Dict:
+    try:
+        with open(kubeconfig) as f:
+            kube_cfg = yaml.safe_load(f)
+            context_names = [n['name'] for n in kube_cfg['contexts']]
+            # namespaces = [n['context']['namespace'] for n in kube_cfg['contexts']]
+            certs = [n['cluster']['certificate-authority-data'] for n in kube_cfg['clusters']]
+            current_context = context or kube_cfg['current-context']
+    except IOError as ioe:
+        print_help_for_missing_or_inaccessible_kubeconfig_file(kubeconfig)
+        sys.exit(-1)
+    except KeyError as ke:
+        print_help_for_bad_kubeconfig_file(kubeconfig)
+        sys.exit(-2)
+
+    try:
+        context_id = context_names.index(current_context)
+    except ValueError:
+        print(f'WARNING: Current context in provided kubeconfig file {kubeconfig} is invalid!')
+        print(f'\nProceeding with explicit context selection....')
+        context_id = select_context_id(kube_cfg)
+
+    defaults = {}
+    defaults['context'] = context_names[context_id]
+    defaults['namespace'] = 'default' # namespaces[context_id]
+    defaults['cert'] = certs[context_id]
+    defaults['config'] = kubeconfig
+    defaults['user'] = 'spark'
+
+    return defaults
+
+def set_up_user(username: str, name_space: str, defaults: Dict) -> None:
+    namespace = name_space or defaults['namespace']
+    kubeconfig = defaults['config']
+    context_name = defaults['context']
+
+    rolebindingname = username + '-role'
+    roleaccess='view'
+    os.system(f"{KUBECTL_CMD} create serviceaccount --kubeconfig={kubeconfig} --context={context_name} {username} --namespace={namespace}")
+    os.system(f"{KUBECTL_CMD} create rolebinding --kubeconfig={kubeconfig} --context={context_name} {rolebindingname} --role={roleaccess}  --serviceaccount={namespace}:{username} --namespace={namespace}")
+
```

Comment:
please also add lines below
```
    os.system(f"{KUBECTL_CMD} label serviceaccount --kubeconfig={kubeconfig} --context={context_name} {username} app.kubernetes.io/managed-by=spark-client --namespace={namespace}")
    os.system(f"{KUBECTL_CMD} label rolebinding --kubeconfig={kubeconfig} --context={context_name} {rolebindingname} app.kubernetes.io/managed-by=spark-client --namespace={namespace}")
```

it will allow us in the future get service account name dynamically, with the following command
```
kubectl --kubeconfig={kubeconfig} --context={context_name} --namespace={namespace} get serviceaccount -l app.kubernetes.io/managed-by=spark-client -o jsonpath={.items[0].metadata.name}
```